### PR TITLE
[DEL-5354]: make resources name unique for upgrade to work properly

### PR DESCRIPTION
**Pull request by AnupamIO**
Testing:
1- Bought up a delegate and upgraded it.
No resources were deleted while upgrading delegate.

Note: one cannot install multiple delegate in same namespace if they are using delegate-service. That's a restriction because of CI module. If anyone wants to install multiple delegates on same namespace, it can be done only if they don't use CI as they can remove delegate-service from manifest.

### CODE REVIEW

**Comment by mdmilic** [comment]
Can you put some description on how this solves the problem. A bit hard to follow it. If I understand `harness-delegate-ng.fullname ` would be delegate name unless overriden. So how does prefixing everything with a delegate name help? I went through the ticket as well, but still not 100% clear. 
Maybe also posting sections of generated yaml would make it easier to understand.

**Comment by AnupamIO** [comment]
> > Can you put some description on how this solves the problem. A bit hard to follow it. If I understand `harness-delegate-ng.fullname ` would be delegate name unless overriden. So how does prefixing everything with a delegate name help? I went through the ticket as well, but still not 100% clear. Maybe also posting sections of generated yaml would make it easier to understand.
> 
> > Can you put some description on how this solves the problem. A bit hard to follow it. If I understand `harness-delegate-ng.fullname ` would be delegate name unless overriden. So how does prefixing everything with a delegate name help? I went through the ticket as well, but still not 100% clear. Maybe also posting sections of generated yaml would make it easier to understand.
> 
> synced offline with Marko, suggestion is to use something like Release.IsUpgrade in condition instead of making unique role and roleBinding, while using Release.IsUpgrade we would need to see how uninstall or rollbacks would work

Few things:
1- It's best practice to keep resource names unique in helm, checked some famous helm charts from artifact hub. ex: https://artifacthub.io/packages/helm/prometheus-community/kube-prometheus-stack and all have release-name prefixed as part of every resource (including role, roleBinding, serviceAccount and all).

2- Checked couple of other things, checked if Release.IsUpgrade can work but it does create issue on upgrade.

3- Other thing we can do is to mark common resources as "helm.sh/resource-policy": keep so these will not be deleted after uninstall or upgrade. That will essentially mean that these resources will never be deleted and if parent release is deleted these will be orphaned in cluster and helm will not manage them.

I would prefer to have unique names only because that's the clearer solution.

**Review by mdmilic — DISMISSED** [review]

**Review by gaurav-jain1 — APPROVED** [review]
